src/mono/browser/runtime/invoke-js.ts

```diff
@@ -17,12 +17,12 @@ import { mono_log_debug, mono_wasm_symbolicate_string } from "./logging";
 import { mono_wasm_get_jsobj_from_js_handle } from "./gc-handles";
 import { endMeasure, MeasuredBlock, startMeasure } from "./profiler";
 import { wrap_as_cancelable_promise } from "./cancelable-promise";
-import { assert_synchronization_context } from "./pthreads/shared";
 
 export const js_import_wrapper_by_fn_handle: Function[] = <any>[null];// 0th slot is dummy, main thread we free them on shutdown. On web worker thread we free them when worker is detached.
 
 export function mono_wasm_bind_js_import(signature: JSFunctionSignature, is_exception: Int32Ptr, result_address: MonoObjectRef): void {
-    assert_bindings();
+    if (MonoWasmThreads) return;
```

Comment:
this seems like it should throw, but i'm guessing it has to be 'return' for tree shaking to work?

Reply:
we have many such places, this is for consistency. Perhaps we could improve that later